main.py

```diff
@@ -515,72 +527,36 @@ def main(cl_arguments):
-                # training since we specifically finetune for each task.
-                pre_target_train = get_best_checkpoint_path(args.run_dir, "pretrain")
-                load_model_state(
-                    model, pre_target_train, args.cuda, skip_task_models=[], strict=strict
-                )
-            else:  # args.transfer_paradigm == "frozen":
-                # Load the current overall best model.
-                layer_path = get_best_checkpoint_path(args.run_dir, "target_train", task.name)
-                assert layer_path, "No best checkpoint found."
-                load_model_state(
-                    model,
-                    layer_path,
-                    args.cuda,
-                    strict=strict,
-                    skip_task_models=task_names_to_avoid_loading,
-                )
-
     if args.do_full_eval:
-        # Evaluate
         log.info("Evaluating...")
         splits_to_write = evaluate.parse_write_preds_arg(args.write_preds)
-        if args.transfer_paradigm == "finetune":
-            if args.do_target_task_training:
-                for task in target_tasks:
-                    ckpt_path = get_best_checkpoint_path(args.run_dir, "target_train", task.name)
-                    assert ckpt_path is not None and ".best" in ckpt_path
-                    load_model_state(
-                        model, ckpt_path, args.cuda, skip_task_models=[], strict=strict
-                    )
-
-                    evaluate_and_write(args, model, [task], splits_to_write)
-            elif args.do_pretrain:
-                # If args.do_target_task_training = 0 and args.do_pretrain = 1
-                # then evaluate on pretraining checkpoints.
-                for task in pretrain_tasks:
-                    ckpt_path = get_best_checkpoint_path(args.run_dir, "pretrain", task.name)
-                    assert ckpt_path is not None and ".best" in ckpt_path
-                    load_model_state(
-                        model, ckpt_path, args.cuda, skip_task_models=[], strict=strict
-                    )
-                    evaluate_and_write(args, model, [task], splits_to_write)
-
-        elif args.transfer_paradigm == "frozen":
-            # Don't do any special checkpointing logic here
-            # since model already has all the trained task specific modules.
-            evaluate_and_write(args, model, target_tasks, splits_to_write)
+        if args.do_target_task_training or (args.load_model and not args.do_pretrain):
+            # If we either do target task training, or if we only evaluate
+            # without pretraining or target task training
+            # then we evaluate on the target tasks.
+            for task in target_tasks:
+                # Find the task-specific best checkpoint to evaluate on.
+                ckpt_path = get_best_checkpoint_path(args, "eval", task.name)
+                assert ckpt_path is not None and ".best" in ckpt_path
```

Comment:
The .best clause here will fail if someone manually specifies a non-best checkpoint, even though there are legitimate use cases for that. Just assert that the path is not none.